newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -401,10 +425,16 @@ export default function SubscriptionDialog({
                   actions = [
                     <FlatButton
                       key="cancel"
-                      disabled={isLoading}
+                      disabled={isLoading || willCancelAtPeriodEnd}
                       label={
                         <LeftLoader isLoading={isLoading}>
-                          <Trans>Cancel your subscription</Trans>
+                          {willCancelAtPeriodEnd ? (
+                            <Trans>
+                              Already cancelled - will expire in the future
```

Comment:
A bit mysterious. Is it possible to be more accurate?

Reply:
Not really because the backend does not say anything.
We could in the future gives a "periodEndsAt" or something similar, but this implies tracking it according to the payment provider... But better than not saying anything.